eclair-core/src/main/scala/fr/acinq/eclair/wire/internal/channel/version0/ChannelTypes0.scala

```diff
@@ -0,0 +1,94 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package fr.acinq.eclair.wire.internal.channel.version0
+
+import fr.acinq.bitcoin.{ByteVector32, OutPoint, Satoshi, Transaction, TxOut}
+import fr.acinq.eclair.channel
+import fr.acinq.eclair.transactions.Transactions._
+
+private[channel] object ChannelTypes0 {
+
+  // The format of the XxxCommitPublished types was changed in version2 to work with anchor outputs channels.
+  // Before that, all closing txs were generated once (when we detected the force-close) and never updated afterwards
+  // (with the exception of 3rd-stage penalty transactions for revoked commitments when one of their htlc txs wins the
+  // race against our htlc-penalty tx, but if that happens a `WatchSpent` will be triggered and we will claim it correctly).
+  // When migrating from these previous types, we can safely set dummy values in the following fields:
+  //  - we only use the `tx` field of `TransactionWithInputInfo` -> no need to completely fill the `InputInfo`
+  //  - `irrevocablySpent` now contains the whole transaction (previously only the txid): we can easily set these when
+  //    one of *our* transactions confirmed, but not when a *remote* transaction confirms. This can only happen for HTLC
+  //    outputs and in these cases we simply remove the entry in `irrevocablySpent`: the channel will set a `WatchSpent`
+  //    which will immediately be triggered and that will let us store the information in `irrevocablySpent`.
+  //  - the `htlcId` in htlc txs is used to detect timed out htlcs and relay them upstream, but it can be safely set to
+  //    0 because the `timedOutHtlcs` in `Helpers.scala` explicitly handle the case where this information is unavailable.
+
+  private def getPartialInputInfo(parentTx: Transaction, childTx: Transaction): InputInfo = {
+    // When using the default commitment format, spending txs have a single input. These txs are fully signed and never
+    // modified: we don't use the InputInfo in closing business logic, so we don't need to fill everything.
+    val input = childTx.txIn.head.outPoint
+    InputInfo(input, parentTx.txOut(input.index.toInt), Nil)
+  }
+
+  case class LocalCommitPublished(commitTx: Transaction, claimMainDelayedOutputTx: Option[Transaction], htlcSuccessTxs: List[Transaction], htlcTimeoutTxs: List[Transaction], claimHtlcDelayedTxs: List[Transaction], irrevocablySpent: Map[OutPoint, ByteVector32]) {
+    def migrate(): channel.LocalCommitPublished = {
+      val htlcTxs = htlcSuccessTxs ++ htlcTimeoutTxs
+      val knownTxs: Map[ByteVector32, Transaction] = (commitTx :: claimMainDelayedOutputTx.toList ::: htlcTxs ::: claimHtlcDelayedTxs).map(tx => tx.txid -> tx).toMap
+      val irrevocablySpentNew = irrevocablySpent.collect { case (outpoint, txid) if knownTxs.contains(txid) => (outpoint, knownTxs(txid)) }
```

Comment:
We can't, it's ok if `irrevocablySpent` contains txs that are not ours (it could be HTLCs that they claimed before we could).

It's also ok that we're losing information about them after this call to `migrate`: we will put a watch-spent at start-up which will instantly trigger and gives us the spending tx, that we will insert into the migrated `irrevocablySpent`.